Boot: Minor fixes for Wii homebrew

leoetlino opened the pull request:
* *Fix BAT setup in Wii mode*: HID4.SBE must be set before calling PowerPC::Update{D,I}BAT, otherwise extended BATs will not be enabled.

* *Change default register values*: Set HID0, HID4, GPR1 to values that are used by libogc for initialisation. This makes boots more similar to a launch from the HBC or another loader, since normally the registers have already been initialised by the loader.

  This fixes a crash in homebrew that assume GPR1 points to a correct location and attempt to use it before initialising registers.

* *Add default exception handlers*: Some homebrew expect exception handlers to be present -- which is almost always the case on console, since most of the time homebrew are launched from either a libogc or SDK title) -- and break if they are not. To fix this, we just need to include default, dummy handlers.

sepalani commented on Source/Core/Core/Boot/Boot.cpp:
```diff
@@ -282,6 +282,18 @@ static void SetDefaultDisc()
+                                                 0x00000900, 0x00000C00, 0x00000D00, 0x00000F00,
+                                                 0x00001300, 0x00001400, 0x00001700};
+
+  const u32 rti_instruction = 0x4C000064;
```
Small nitpick, I'm pretty sure Return From Interrupt is called rfi for PowerPC architecture.

leoetlino replied:
Fixed, thanks!

Ebola16 commented:
If you're looking for another case of Wii homebrew issues, this PR doesn't fix [Aspirin homebrew app causes Dolphin to become unresponsive on startup](https://bugs.dolphin-emu.org/issues/9768)

leoetlino commented:
Not in the scope of this PR (which addresses issues that affect Wii homebrew that are already detected as such), but I've updated the issue you linked.
